defold-spine/commonsrc/vertices.cpp

```diff
@@ -169,22 +188,27 @@ static void CalcAndAddVertexBufferAttachmentByClipper(dmArray<float>& scratch, s
     else if (attachment->type == SP_ATTACHMENT_MESH)
     {
         spMeshAttachment *mesh = (spMeshAttachment *) attachment;
+        attachmentColor = &mesh->color;
+        if (attachmentColor->a == 0)
+        {
+            spSkeletonClipping_clipEnd(skeleton_clipper, slot);
+            return;
+        }
         EnsureArraySize(scratch, mesh->super.worldVerticesLength);
         spVertexAttachment_computeWorldVertices(SUPER(mesh), slot, 0, mesh->super.worldVerticesLength, scratch.Begin(), 0, 2);
 
-        vertex_count  = mesh->super.worldVerticesLength >> 1;
+        vertex_count  = mesh->super.worldVerticesLength;
         uvs           = mesh->uvs;
         indices       = mesh->triangles;
         indices_count = mesh->trianglesCount;
         vertices      = scratch.Begin();
     }
 
-    spSkeletonClipping_clipTriangles(skeleton_clipper, vertices, vertex_count << 1, indices, indices_count, uvs, 2);
-    vertex_count  = skeleton_clipper->clippedVertices->size >> 1;
+    spSkeletonClipping_clipTriangles(skeleton_clipper, vertices, vertex_count, indices, indices_count, uvs, 2);
     indices_count = skeleton_clipper->clippedTriangles->size;
 
     *out_count             += indices_count;
-    *out_max_triangle_count = dmMath::Max(*out_max_triangle_count, indices_count * 2);
+    *out_max_triangle_count = dmMath::Max(*out_max_triangle_count, vertex_count);
```

Comment:
This part is the actual fix: to be able to clip Triangles we need to know maximal `worldVerticesLength`

Reply:
Imho, `out_max_triangle_count` is badly named. It's not really counting any triangles, it's the number of vertices (or more accurately number of floats in the array)

And, `out_count` is also not great.

Perhaps rename to `out_count` -> `out_indices` and `out_max_triangle_count` -> `out_vertices` ?

Reply:
Fixed!